include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/GenerateComposePreviewRobolectricTestsExtension.kt

```diff
@@ -209,24 +235,82 @@ abstract class GenerateComposePreviewRobolectricTestsTask : DefaultTask() {
   @get:Input
   abstract val robolectricConfig: MapProperty<String, String>
 
+  @get:Input
+  abstract val numOfShards: Property<Int>
+
   @TaskAction
   fun generateTests() {
     val testDir = outputDir.get().asFile
     testDir.mkdirs()
 
     val packagesExpr = scanPackageTrees.get().joinToString(", ") { "\"$it\"" }
     val includePrivatePreviewsExpr = includePrivatePreviews.get()
+    val numShards = numOfShards.get()
+
+    require(numShards >= 1) {
+      "numOfShards must be >= 1, but was $numShards"
+    }
 
     val generatedClassFQDN = "com.github.takahirom.roborazzi.RoborazziPreviewParameterizedTests"
     val packageName = generatedClassFQDN.substringBeforeLast(".")
-    val className = generatedClassFQDN.substringAfterLast(".")
+    val baseClassName = generatedClassFQDN.substringAfterLast(".")
     val directory = File(testDir, packageName.replace(".", "/"))
     directory.mkdirs()
+
+    // Delete old generated test files to avoid conflicts when changing numOfShards
+    directory.listFiles()?.filter { it.extension == "kt" }?.forEach { it.delete() }
     val robolectricConfigString =
       "@Config(" + robolectricConfig.get().entries.joinToString(", ") { (key, value) ->
         "$key = $value"
       } + ")"
     val testerQualifiedClassNameString = testerQualifiedClassName.get()
+
+    if (numShards == 1) {
```

Comment:
Minor: I'd also handle the edge case `numOfShards <= 0` and throw the corresponding exception